### PR TITLE
UXD-843 fix(ListBoxWithTags): fix accessibility issues

### PR DESCRIPTION
**Pull request by oscarkwan**
### Purpose 🚀
- Fix accessibility issues with ListBoxWithTags

### Notes ✏️
- add `aria-expanded` and `aria-haspopup` to Trigger
- add aria label text to close button
- make tag close button keyboard accessible

Resolved https://github.com/acl-services/paprika/issues/820

### Updates 📦
- [x] MINOR (backward compatible) change to **Tag** **ListBoxWithTags**
- [x] PATCH (bug fix) change to _these packages_ **OverflowMenu** **Confirmation**

### Storybook 📕
http://storybooks.highbond-s3.com/paprika/UXD-843-listbox-tags-a11y

### CODE REVIEW

File: packages/ListBox/src/components/Options/helpers/options.js
Comment (oscarkwan) [inline packages/ListBox/src/components/Options/helpers/options.js:204]: https://github.com/acl-services/paprika/pull/941 has the same change, will wait for that to get merged

File: packages/Tag/src/Tag.js
Comment (oscarkwan) [inline packages/Tag/src/Tag.js:18]: I Included backspace, but open to not including this.

File: packages/ListBoxWithTags/src/ListBoxWithTags.js
Comment (mikrotron) [inline packages/ListBoxWithTags/src/ListBoxWithTags.js:78]: Do you think we should also have `aria-controls` here?
https://www.w3.org/TR/wai-aria-1.1/#aria-controls

Comment (mikrotron) [reply packages/ListBoxWithTags/src/ListBoxWithTags.js:78]: We do use it here:
https://github.com/acl-services/paprika/blob/master/packages/OverflowMenu/src/components/Trigger/Trigger.js#L20

Comment (oscarkwan) [reply packages/ListBoxWithTags/src/ListBoxWithTags.js:78]: Yeah, lets add it

File: packages/ListBoxWithTags/src/ListBoxWithTags.js
Comment (mikrotron) [inline packages/ListBoxWithTags/src/ListBoxWithTags.js:76]: The default label for this trigger button gets unnecessarily long and redundant because it includes the desciption of all of the remove buttons too.  I thin we should mitigate that by providing an `aria-label` (`a11yText`) that can be composed more thoughtfully.

File: packages/Tag/src/Tag.js
Comment (mikrotron) [reply packages/Tag/src/Tag.js:18]: This is the only place I'm aware of where the `backspace` key will activate a remove button... but I like it!

But does @sunminsarahlee approve? This means when the keyboard focus is on the little `x` for a selected tag, pressing any of these keys removes it.

File: packages/Tag/src/Tag.js
Comment (mikrotron) [inline packages/Tag/src/Tag.js:29]: What if `children` is a `node`, not a `string`?
Maybe it should fall back to the generic "remove"?

File: packages/Tag/src/Tag.js
Comment (tristanjasper) [reply packages/Tag/src/Tag.js:18]: I would think just "Enter" or "Space" would be the expected buttons to press to action this imo

File: packages/ListBoxWithTags/src/ListBoxWithTags.js
Comment (tristanjasper) [inline packages/ListBoxWithTags/src/ListBoxWithTags.js:77]: just `aria-haspopup`. No need to assign to "true"

File: packages/Tag/src/Tag.js
Comment (tristanjasper) [inline packages/Tag/src/Tag.js:17]: perhaps name this function something that associates it with the Delete Button, maybe `handleDeleteKeyDown`?
Maybe Delete Component should be moved to its own file with export at this point

File: packages/ListBoxWithTags/src/ListBoxWithTags.js
Comment (oscarkwan) [reply packages/ListBoxWithTags/src/ListBoxWithTags.js:77]: Actually no because `aria-haspopup` expects a string of "true" no the actual value of true

Comment (tristanjasper) [reply packages/ListBoxWithTags/src/ListBoxWithTags.js]: Not sure what you mean, both ways ends up adding `aria-haspopup="true"` to the dom

Comment (oscarkwan) [reply packages/ListBoxWithTags/src/ListBoxWithTags.js]: Hm, i wonder why we use `aria-haspopup="true"` in 3 other places then

Comment (tristanjasper) [reply packages/ListBoxWithTags/src/ListBoxWithTags.js]: Perhaps update the 3 other places as well if it's not necessary? it's a small cleanup anyway

File: packages/Confirmation/src/components/TriggerButton/TriggerButton.js
Comment (mikrotron) [inline packages/Confirmation/src/components/TriggerButton/TriggerButton.js:23]: This change and the one to `OverflowMenu` are patch updates and while we _could_ include them in the `CHANGELOG`, I think they're so small and insignificant that we can omit that update.

File: packages/Tag/src/Tag.js
Comment (mikrotron) [inline packages/Tag/src/Tag.js:1]: Your changes to the Tag component though are more significant and should be added to the `CHANGELOG`.

File: packages/ListBoxWithTags/CHANGELOG.md
Comment (mikrotron) [inline packages/ListBoxWithTags/CHANGELOG.md:10]: This is an internal implementation detail that I don't think needs to be logged.  🤷‍♂️

**Review by mikrotron — APPROVED** [review]
llgtm 07

**Pull request merged**
